book/src/dev/rfcs/0005-state-updates.md

```diff
@@ -421,22 +668,24 @@ Returns
 
 Implemented by querying:
 
-- (non-finalized) XXX parts of the non-finalized state;
+- (non-finalized) the `height_by_hash` of each chain starting with the best
+  chain, then find block in `blocks` of that chain.
 - (finalized) the `height_by_hash` (to get the block height) and then
     `block_by_height` (to get the block data) trees.
 
 
 # Drawbacks
 [drawbacks]: #drawbacks
 
-# Rationale and alternatives
-[rationale-and-alternatives]: #rationale-and-alternatives
+- Restarts can cause `zebrad` to redownload up to the last one hundred blocks
+  it verified.
 
-# Prior art
-[prior-art]: #prior-art
+- The service interface puts some extra responsibility on callers to ensure
+  it is used correctly and does not verify the usage is correct at compile
+  time.
```

Comment:
What responsibility, exactly? Is this referring to the `CommitFinalizedBlock`/`CommitBlock` distinction, or something else?

Reply:
this is referring to the `ready()` before each `call()` required by Tower

Reply:
I would say both